### PR TITLE
Starting multiple gcsfuse processes using PAPI action list

### PR DESCRIPTION
**Pull request by samanvp**
In the latest version of DeepVariant (0.8.0) gcsfuse binary has been removed from the docker image. In order to be able to run make_example on multi core machines we still need to start multiple gcsfuse processes (one per core).
We achieve this by launching PAPI using a json file that includes [action list](https://github.com/googlegenomics/pipelines-tools/blob/749315a73e6c3bd5277351e32a365f42198db1ae/pipelines/internal/commands/run/run.go#L21) in the following order:

- mkdir local directories for gcsfuse
- gcsfuse process 1
- gcsfuse process 2
- ...
- gcsfuse process n
- seq 1 n | parallel make_example

This will be essentially equivalent to what we were doing using DeepVariant 0.7.2

### CODE REVIEW

File: gcp_deepvariant_runner.py [hunk 1]
```diff
@@ -210,17 +195,72 @@ def _get_base_job_args(pipeline_args):
...
+    is_gcsfuse_activated: whether or not read input bam file using gcsfuse.
+    deep_variant_image: DeepVariant image given using --docker_image flag.
+    make_example_command_template: template command used in actions list.
+  """
+  gcs_bucket = _get_gcs_bucket(input_bam_file)
+  bam_file_relative_path = _get_gcs_relative_path(input_bam_file)
+
+  actions = []
+  if is_gcsfuse_activated:
+    io_command = 'mkdir -p'
```
Comment (kemp-google) [inline gcp_deepvariant_runner.py]: mkdir_command?

File: gcp_deepvariant_runner.py [hunk 2]
```diff
@@ -210,17 +195,72 @@ def _get_base_job_args(pipeline_args):
...
+    for shard_index in range(shard_start_index, shard_end_index + 1):
+      local_dir = GCSFUSE_LOCAL_DIR_TEMPLATE.format(SHARD_INDEX=shard_index)
+      io_command += ' ' + local_dir
+      gcs_fuse_command = GCSFUSE_COMMAND_TEMPLATE.format(
+          BUCKET=gcs_bucket, LOCAL_DIR=local_dir)
+      actions.append({'imageUri': GCSFUSE_IMAGE,
```
Comment (kemp-google) [inline gcp_deepvariant_runner.py]: You could just create the directories before running fuse (so, eg, each action is now basically mkdir mount point; start gcsfuse on mount point

File: gcp_deepvariant_runner.py [hunk 3]
```diff
@@ -210,17 +195,72 @@ def _get_base_job_args(pipeline_args):
...
+                          bam_file_relative_path)
+  else:
+    local_bam_template = '$INPUT_BAM'
+
+  make_example_command = make_example_command_template.format(
+      SHARD_START_INDEX=shard_start_index, SHARD_END_INDEX=shard_end_index,
+      TASK_INDEX='{}', INPUT_BAM=local_bam_template)
+  actions.append(
+      {'imageUri': deep_variant_image,
+       'commands': ['-c', make_example_command],
```
Comment (kemp-google) [inline gcp_deepvariant_runner.py]: Before starting this you need to wait for the mount points to be available.

File: gcp_deepvariant_runner.py (hunk 1 above)
Comment (samanvp) [reply gcp_deepvariant_runner.py]: done.

**Comment by samanvp** [comment]
I ran a bunch of profiling experiments to compare this new implementation of gcsfuse to our current one (which gcsfuse binary exists in deepvariant docker image) and the performance is almost identical.

File: gcp_deepvariant_runner_test.py [hunk 4]
```diff
@@ -211,12 +211,15 @@ def testRunPipeline_WithGVCFOutFile(self, mock_can_write_to_bucket,
...
-  def testRunMakeExamples(self, mock_can_write_to_bucket, mock_obj_exist,
-                          mock_pool):
+  @mock.patch('tempfile.NamedTemporaryFile')
+  def testRunMakeExamples(self, mock_temp_file, mock_can_write_to_bucket,
+                          mock_obj_exist, mock_pool):
     mock_apply_async = mock_pool.return_value.apply_async
     mock_apply_async.return_value = None
     mock_obj_exist.return_value = True
     mock_can_write_to_bucket.return_value = True
+    # TODO(samanvp): it will be a stronger test if we validate content of file.
```
Comment (kemp-google) [inline gcp_deepvariant_runner_test.py]: TODOs should point to an issue.

eg

TODO(#1234): Update test to validate the content of the file.

And in that issue, explain why it is too hard to just do now (or just do it now :)

Comment (samanvp) [reply gcp_deepvariant_runner_test.py]: I removed the TODOs and added the json files verification to the unit tests.

**Pull request merged**